feat(MA): add campaign field preferences id or campaign name

jabahamondes opened the pull request:
## Problem

Users usually use campaign name (us) or campaign id as utm_campaign. So we can, by the settings allow the user to pick one or the other per integration. This will use automatically one or the other field without them to manually map the id to the campaign.

## Changes
This change improves flexibility in campaign tracking and enhances the overall marketing analytics capabilities.

- Introduced CampaignFieldPreference schema to manage campaign matching fields (campaign_name or campaign_id)
- Updated MarketingAnalyticsConfig to include campaign_field_preferences
- Added id field to marketing analytics schema
- Implemented UI component for configuring campaign field preferences
- Enhanced backend logic to support new preferences and ensure validation
- Updated relevant adapters to handle campaign ID alongside name in queries

## How did you test this code?
Updated test and manually
Note: the ui will be updated in a following PR:
<img width="921" height="620" alt="image" src="https://github.com/user-attachments/assets/d0c21732-6ea3-4db4-9757-f643ac61eae1" />


We also have the id and campaign name in the table now:
<img width="318" height="305" alt="image" src="https://github.com/user-attachments/assets/54ebd7da-b76a-4a3b-9ccd-fd9e8c8c0f3d" />

Commit:
feat(marketing-analytics): add custom UTM source mappings configuration

- Introduced CustomSourceMappingsConfiguration component for managing custom UTM source mappings in marketing analytics settings.
- Updated MarketingAnalyticsSettings to include the new component.
- Enhanced marketingAnalyticsSettingsLogic to handle custom source mappings.
- Added database migration for custom_source_mappings field in TeamMarketingAnalyticsConfig.
- Implemented validation for custom source mappings to ensure uniqueness across integrations.
- Updated MarketingSourceFactory to merge custom mappings with adapter defaults.
- Added integration tests for custom source mappings functionality.

Commit:
feat(marketing): add campaign field preferences and ID handling in marketing analytics

- Introduced CampaignFieldPreference schema to manage campaign matching fields (campaign_name or campaign_id).
- Updated MarketingAnalyticsConfig to include campaign_field_preferences and custom_source_mappings.
- Added id field to marketing analytics schema for better tracking.
- Implemented UI component for configuring campaign field preferences.
- Enhanced backend logic to support new preferences and ensure validation.
- Updated relevant adapters to handle campaign ID alongside name in queries.

This change improves flexibility in campaign tracking and enhances the overall marketing analytics capabilities.

Commit:
Merge conflicts

github-actions[bot] committed:
Update query snapshots

Commit:
Fixing error

Commit:
Merge origin/master and resolve migration conflict

lricoy reviewed (approved):
🚢 , but I think the bot is right

Commit:
feat(tests): update campaign field preferences validation and add BingAds adapter tests

- Simplified campaign field preferences structure by removing fallback fields in tests.
- Added tests to ensure BingAds adapter respects campaign field preferences and always returns both name and id fields.
- Updated group by expressions in BingAds adapter to include campaign id.

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-utm-campaign-configuration-id-or-name

Commit:
Fixing mypybaseline

Commit:
Fixing tests

Commit:
Merge remote-tracking branch 'origin/master' into feat/adding-utm-campaign-configuration-id-or-name

Commit:
Fixing tests

Commit:
Fixing tests

jabahamondes committed:
Merge branch 'master' into feat/adding-utm-campaign-configuration-id-or-name

jabahamondes committed:
Merge branch 'master' into feat/adding-utm-campaign-configuration-id-or-name

jabahamondes committed:
Merge branch 'master' into feat/adding-utm-campaign-configuration-id-or-name

jabahamondes committed:
Fixing test

jabahamondes committed:
Fixing snapshot

github-actions[bot] committed:
Update query snapshots
